### ISSUE TITLE
Optionally use local IP when deployed in AWS

### ISSUE THREAD

**Issue by tbfangel** [jquatier/eureka-js-client#51]
Hi,

We use eureka-js-client in our AWS deployment where all our services are accessed through zuul and zuul should be the only component deployed with a public IP. Currently, in an AWS setup the ip and hostname used to register in Eureka is automatically pulled from the AWS metadata `public-ipv4` and `public-hostname` in the function `addInstanceMetadata`. This requires us right now to deploy our node.js service in an instance with a public IP. 

We would like to use the local hostname and ip instead. A solution would be to add a configuration option telling the client to use local instead of public metadata for both the ip and hostname config and also for replacement of the token `__HOST__` in healthCheckUrl and statusPageUrl.

BR,
/tbfangel

**Comment by jquatier** [comment]
That makes sense. When I wrote the metadata fetching code I copied what the Java client does, which is grab the public hostname. See `getHostName' here: https://github.com/Netflix/eureka/blob/master/eureka-client/src/main/java/com/netflix/appinfo/CloudInstanceConfig.java

Anyway, I understand the need and would gladly accept a PR with a `useLocalMetadata` option and a unit test if you're interested. Otherwise, I can take a look when I get some time. Thanks for the report and glad to hear this module is working for your project!
